### PR TITLE
[AIFBR-7974] Add registry export/import scripts

### PR DESCRIPTION
**Pull request by vivekjhauipath**
- Add registry/export import scripts which need to create a nodeport for registry
- Move package import/export to folder
- Separate out token fetch code for reuse

### CODE REVIEW

File: platform/onebox/backup_and_restore/ceph/get-credentials.sh [hunk 1]
```diff
@@ -22,6 +22,7 @@ function initialize_variables() {
     else
     	OBJECT_GATEWAY_EXTERNAL_HOST=$PRIVATE_IP
     fi
+    echo "$green $(date) Private IP was $PRIVATE_IP and OBJECT_GATEWAY_EXTERNAL_HOST is $OBJECT_GATEWAY_EXTERNAL_HOST"
```
Comment (SahilChaudhary25) [inline platform/onebox/backup_and_restore/ceph/get-credentials.sh:25]: Private IP is

File: platform/onebox/backup_and_restore/registry/export.sh [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+function get_image_list() {
+	# sqlcmd -S tcp:${DB_CONN} -d ${DB_NAME} -U ${DB_USER} -P ${DB_PASSWORD} -i getimages.sql -o images.txt -h -1 -W
+	sqlcmd -S tcp:${DB_CONN} -d ${DB_NAME} -U ${DB_USER} -P ${DB_PASSWORD} -o images.txt -h -1 -W -Q "set nocount on; select distinct mpi.image_uri from ml_package_images mpi inner join ml_skill_versions msv on msv.ml_package_version_id = mpi.version_id and  msv.processor = mpi.processor where msv.status in ('UPDATING', 'COMPLETED', 'VALIDATING_DEPLOYMENT') and mpi.status = 'ACTIVE'"
+}
+
+formulate_docker_command() {
+    docker images registry
+    if [[ $? -ne 0 ]]; then
+        echo "sudo permission required for docker"
```
Comment (SahilChaudhary25) [inline platform/onebox/backup_and_restore/registry/export.sh:77]: It could be failed due to other issues rather than sudo permission

File: platform/onebox/backup_and_restore/registry/export.sh [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+
+validate_input
+
+docker_setup
+
+#validate_file_path "./is.sh"
+#source ./is.sh
+#
+## register
+#register_client_and_fetch_access_token
```
Comment (SahilChaudhary25) [inline platform/onebox/backup_and_restore/registry/export.sh:148]: can we remove commented code

File: platform/onebox/backup_and_restore/registry/get-credentials.sh [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+
+function get_db_details() {
+	## get password for deployer db => Using Deployer pod
+	#line=$(kubectl get secrets -n aifabric ai-deployer-secrets -o yaml | grep 'DATASOURCE_PASSWORD')
+	#readonly DB_PASSWORD=$(echo ${line##* } | base64 -d)
+	## get url
+	#line=$(kubectl -n aifabric get deployment ai-deployer-deployment -o yaml | grep -A1 'SPRING_DATASOURCE_URL' | grep -v 'SPRING_DATASOURCE_URL')
+	#readonly DB_CONN=${line##* }
+	## get user
+	#line=$(kubectl -n aifabric get deployment ai-deployer-deployment -o yaml | grep -A1 'SPRING_DATASOURCE_USERNAME' | grep -v 'SPRING_DATASOURCE_USERNAME')
```
Comment (SahilChaudhary25) [inline platform/onebox/backup_and_restore/registry/get-credentials.sh:42]: same here

**Review by SahilChaudhary25 — CHANGES_REQUESTED** [review]
Please add docs to function also

File: platform/onebox/backup_and_restore/registry/export.sh (hunk 2 above)
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/registry/export.sh:77]: existing command used in other scripts

File: platform/onebox/backup_and_restore/ceph/get-credentials.sh (hunk 1 above)
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/ceph/get-credentials.sh:25]: will handle later. nit.

File: platform/onebox/backup_and_restore/registry/export.sh (hunk 3 above)
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/registry/export.sh:148]: ok

File: platform/onebox/backup_and_restore/registry/export.sh [hunk 5]
```diff
@@ -0,0 +1,146 @@
+#!/bin/bash
```
Comment (rajivml) [inline platform/onebox/backup_and_restore/registry/export.sh:1]: rename the file to a better name like export_images.sh

File: platform/onebox/backup_and_restore/registry/export.sh [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+
+validate_setup
+
+validate_input
+
+docker_setup
+
+get_image_list
+
+save_images
```
Comment (rajivml) [inline platform/onebox/backup_and_restore/registry/export.sh:146]: at the start of the script, check if sufficient space is available on the disk to export the images and also recommend users to use a secondary disk which is non root disk to avoid kubernetes getting impacted because of disk space issue

File: platform/onebox/backup_and_restore/registry/get-credentials.sh [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+	line=$(kubectl get jobs ${provisionPod} -o yaml | grep -A1 'DEPLOYER_DBNAME' | grep -v 'DEPLOYER_DBNAME')
+	readonly DB_NAME=${line##* }
+
+	if [[ -z $DB_CONN || -z $DB_USER || -z $DB_PASSWORD || -z $DB_NAME ]]; then
+    	echo "$red $(date) Failed to fetch one or more db info, Please check ... Exiting $default"
+    	exit 1
+  	fi
+}
+
+function get_registry_details() {
```
Comment (rajivml) [inline platform/onebox/backup_and_restore/registry/get-credentials.sh:52]: indentation of if else statements doesn't look right

File: platform/onebox/backup_and_restore/registry/import.sh [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+    docker images registry
+    if [[ $? -ne 0 ]]; then
+        echo "sudo permission required for docker"
+        DOCKER_COMMAND="sudo docker"
+    else
+        DOCKER_COMMAND="docker"
+    fi
+}
+
+function docker_setup() {
```
Comment (rajivml) [inline platform/onebox/backup_and_restore/registry/import.sh:79]: why is this required and why are we not connecting to docker through user name and password

File: platform/onebox/backup_and_restore/registry/export.sh (hunk 5 above)
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/registry/export.sh:1]: why, we have kept it in separate folders.
So we will have
regsitry/(ex/im)port.sh
storage/(ex/im)port.sh
packages/(ex/im)port.sh
datasets/(ex/im)port.sh

File: platform/onebox/backup_and_restore/registry/export.sh (hunk 6 above)
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/registry/export.sh:146]: @rajivml the script is run on a separate VM. Hence not valid

File: platform/onebox/backup_and_restore/registry/import.sh (hunk 8 above)
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/registry/import.sh:79]: existing scripts behave in similar manner, copied from there

File: platform/onebox/backup_and_restore/registry/export.sh (hunk 6 above)
Comment (rajivml) [reply platform/onebox/backup_and_restore/registry/export.sh:146]: yeah but the path where they are exporting should have at least 50 -100 gigs of free space to export the same otherwise the export might fail in the middle right

File: platform/onebox/backup_and_restore/credentials/is.sh [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+  cookie_file="cookfile.txt"
+  cookie_file_new="cookfile_new.txt"
+
+  # Get token and construct the cookie, save the returned token.
+  curl --silent --fail --show-error -k -c $cookie_file --request GET "$antif"
+
+  # Replace headers
+  sed 's/XSRF-TOKEN-IS/XSRF-TOKEN/g' $cookie_file >$cookie_file_new
+
+  token=$(cat $cookie_file_new | grep XSRF-TOKEN | cut -f7 -d$'\t')
```
[the remark this answers is not in the document]
Comment (vivekjhauipath) [reply platform/onebox/backup_and_restore/credentials/is.sh:43]: curl will fail before this, we are assuming that IDP will behave properly

**Review by SahilChaudhary25 — APPROVED** [review]
LGTM

**Review by rajivml — APPROVED** [review]
